### PR TITLE
[typescript-angular] Add samples with composed schemas

### PR DESCRIPTION
**Pull request by amakhrov**
Composed schemas (e.g. allOf) have been a sensitive area for multiple generators - `typescript-angular` included.
Yet there are no samples to verify the behavior, validate changes introduced by PRs. For example, [this bug report](https://github.com/OpenAPITools/openapi-generator/issues/16150) mentions multiple typescript- generators might be affected - without evidence in the samples

This PR adds samples generated from a spec with various composed schemas.
Only contains allOf so far - which, I believe, is most commonly used. Happy to add other types (e.g. oneOf, examples with more deeply nested inheritance tree) if deemed practical and beneficial.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11) @amakhrov (2020/02) @davidgamero (2022/03) @mkusaka (2022/04)

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR.

Can you please add the new folders to https://github.com/OpenAPITools/openapi-generator/blob/master/pom.xml#L1254 so that CircleCI will test these new samples as part of the build?

Remember to add pom.xml in each sample folders.

**Comment by amakhrov** [comment]
@wing328 could you please clarify the intention here? Do we want to have some tests for the generated samples?
Currently, those samples don't have any tests inside. The idea was that at it would help when reviewing relevant PRs that affect generation of composed models.

I guess for a test, typechecking of the generated models could be added.

**Comment by TiFu** [comment]
Think that would make sense.
In addition to pure compile tests, this could also be extended to test whether  the "right" properties have been generated.

**Comment by wing328** [comment]
@amakhrov I was referring to compile tests, which should help catch the issue reported in https://github.com/OpenAPITools/openapi-generator/issues/16150 moving forward.

**Comment by amakhrov** [comment]
Hmm.
> [ERROR] org.openapitools.codegen.ArchUnitRulesTest.testLoggersAreNotPublicFinalAndNotStatic on testLoggersAreNotPublicFinalAndNotStatic(org.openapitools.codegen.ArchUnitRulesTest)
> Field <org.openapitools.codegen.languages.AbstractAdaCodegen.LOGGER> has modifier STATIC in (AbstractAdaCodegen.java:0)

Don't believe it's caused by this PR

**Comment by amakhrov** [comment]
> Can you please add the new folders to https://github.com/OpenAPITools/openapi-generator/blob/master/pom.xml#L1254 

Done @wing328

File: samples/client/others/typescript-angular/pom.xml
Comment (amakhrov) [inline samples/client/others/typescript-angular/pom.xml:1]: this file is copied over from petstore/typescript-angular-16-provided-in-root, with some cleanup

File: samples/client/others/typescript-angular/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    <groupId>org.openapitools</groupId>
+    <artifactId>typescript-angular-other-composed-schemas</artifactId>
+    <packaging>pom</packaging>
+    <version>1.0-SNAPSHOT</version>
+
+    <name>Typescript-Angular Composed Schemas generation</name>
+    <build>
+        <plugins>
+            <plugin>
+                <artifactId>maven-dependency-plugin</artifactId>
```
Comment (amakhrov) [inline samples/client/others/typescript-angular/pom.xml:13]: wondering whether this plugin is needed at all here

File: samples/client/others/typescript-angular/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  "compileOnSave": false,
+  "include": [
+    "builds/composed-schemas/index.ts",
+    "builds/composed-schemas-tagged-unions/index.ts"
```
Comment (amakhrov) [inline samples/client/others/typescript-angular/tsconfig.json:5]: the both builds are listed here, in a single compilation target

**Review by macjohnny — APPROVED** [review]
thx

**Pull request merged**